client/client.go

```diff
@@ -771,6 +771,11 @@ func (c *Client) NodeID() string {
 	return c.config.Node.ID
 }
 
+// NodeIP returns the IP address the node is listening
```

Comment:
Oh, right... this comment points out a problem with trying to expose the address of the client... the client doesn't bind on the RPC address (which is why you had to thread it down from the agent config). So this might be the right IP address from the configuration but isn't necessarily meaningful to the operator if they've got multiple interfaces on the host. 

I suspect a lot of clients are just going to show 127.0.0.1 here, which isn't terribly useful as a metric. But I think that's _also_ going to be a problem for the HTTP address. Before I send you off on another wild goose chase here, maybe we should ping @mikenomitch and ask for what he was originally thinking with https://github.com/hashicorp/nomad/issues/12921?

Reply:
Sounds good, I will follow-up on that, thanks for the guidance.

Another option that I thought about was to use to first fingerprinted IP, which also what's set in `unique.network.ip-address`, but I don't know if that's a good value either. 

I will leave this open until we get more clarification 👍